File: python/Node.py
class Node:
    def __init__(self, layer, eventSample):
        self.layer = layer
        self.eventSample = eventSample
        self.right = None
        self.left = None
        self.cutValue = None
        self.cutFeature = None
        self.isLeaf = False
        self.class_ = None

    def setLeafClass(self):
        self.isLeaf = True
        numSig = 0
        numBkg = 0
        for sample in self.eventSample:
            if sample[2]==0:
                numSig += 1
            else:
                numBkg += 1
        if numSig>numBkg:
            self.class_ = 0
        else:
            self.class_ = 1

    def cut(self, features):
        if features[self.cutFeature]>self.cutValue:
            return self.right
        else:
            return self.left

File: src/Node.cc
#include <Node.h>

Node::Node(){
    right = 0;
    left = 0;
}

Node* Node::evaluateValues(std::vector<float>* features){
    if(features->at(cutFeature)>cutValue) return right;
    else return left;
}

Node* Node::evaluateBins(std::vector<unsigned int>* bins){
    if(bins->at(cutFeature)>cutBin) return right;
    else return left;
}
